main.tf

```diff
@@ -16,165 +66,184 @@ data "ibm_is_security_group" "existing_sg" {
   name  = var.existing_security_group_name
 }
 
+############################################################################
+
+############################################################################
+# Security Group Locals
+############################################################################
+
+locals {
+  sg_id = var.existing_security_group_name != null ? data.ibm_is_security_group.existing_sg[0].id : ibm_is_security_group.sg[0].id
+}
+
+############################################################################
 
 #############################################################################
 # Security Group Target
 #############################################################################
 
 resource "ibm_is_security_group_target" "sg_target" {
   count          = length(var.target_ids)
-  security_group = var.existing_security_group_name != null ? data.ibm_is_security_group.existing_sg[0].id : ibm_is_security_group.sg[0].id
+  security_group = local.sg_id
   target         = var.target_ids[count.index]
 }
 
-#############################################################################
-# Security Group Rule
 #############################################################################
 
-resource "ibm_is_security_group_rule" "security_group_rule" {
-  count     = length(local.all_rules)
-  group     = var.existing_security_group_name != null ? data.ibm_is_security_group.existing_sg[0].id : ibm_is_security_group.sg[0].id
-  direction = local.all_rules[count.index].direction
-  remote    = local.all_rules[count.index].remote
+##############################################################################
+# Create Rules
+##############################################################################
 
-  dynamic "tcp" {
+resource "ibm_is_security_group_rule" "rule" {
```

Comment:
Since you have renamed the resource from `ibm_is_security_group_rule.security_group_rule` to `ibm_is_security_group_rule.rule` the upgrade pipeline is failing since it is detecting this as a breaking change. Such an update will cause the following resource to be destroyed:

`security_group_rule Address: module.create_sgr_rule.ibm_is_security_group_rule.security_group_rule[3]`
`security_group_rule Address: module.create_sgr_rule.ibm_is_security_group_rule.security_group_rule[0]`
`security_group_rule Address: module.create_sgr_rule.ibm_is_security_group_rule.security_group_rule[1]`
`security_group_rule Address: module.create_sgr_rule.ibm_is_security_group_rule.security_group_rule[2]`
`security_group_rule Address: module.create_sgr_rule1.ibm_is_security_group_rule.security_group_rule[0]`

Is it possible to see if we can prevent a breaking change here? Maybe start by using the same resource name that was already there and we can check the results of the upgrade test?

Reply:
ok changed